test_generated_index.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+output_dir="$1"
+if [ -e "${output_dir}" ]; then
+    echo "ERROR: Output directory exists. This script requires that it not exist already." 1>&2
+    exit 1
+fi
+mkdir -p "${output_dir}"
+
+set -xeo pipefail
+
+# Setup venv for the generator script.
+venv_dir="${output_dir}/.venv"
+python3.9 -m venv "${venv_dir}"
+"${venv_dir}/bin/pip" install -r ./requirements.txt
+
+# Generate the Pants PyPi-compatible index.
+mkdir -p "${output_dir}/public/simple"
+"${venv_dir}/bin/python" ./generate_index.py > "${output_dir}/public/simple/index.html"
+
+# Serve a copy of the generated index on port 8080.
+python3.9 -m http.server -d "${output_dir}/public" --bind 127.0.0.1 8080 &
+
+# Setup another virtual environment in temporary directory and install a version of Pants into it.
+# Note: The Python version (e.g., 3.9) must match the Python version required by the version of Pants
+#  to be installed as a test, or else Pip will otherwise ignore the Pants version and fail.
+pants_venv_dir="${output_dir}/pants-venv"
+python3.9 -m venv "${pants_venv_dir}"
+"${pants_venv_dir}/bin/pip" install -vv \
+  --extra-index-url=http://127.0.0.1:8080/simple/ \
+  pantsbuild.pants==2.18.0a0
```

Comment:
This choice of version is necessary for the test to succeed. Pip apparently ignores the links for newer Pants versions as invalid given our current index format. This will be updated once https://github.com/pantsbuild/wheels.pantsbuild.org/pull/1 lands and the index format is updated to comply with PEP 503.